### ISSUE TITLE
Crash on M1 with standard pipeline

### ISSUE THREAD

**Issue by ryleu** [grondag/canvas#288]
**System Informatinon**

- Minecraft Version: 1.17
- Canvas Version: 1.17-1.0.1641
- Operating System: MacOS v11.4 (Big Sur) on ARM
- Graphics Card: Apple M1
- Java: Zulu 16.0.1+9

**Other mods and versions installed**

- fabric-api-0.36.0+1.17

**How you triggered the behavior**

Joining a world with the pipeline set to "standard"

**What you expected would happen**

Bloom would start working

**What actually happened**

Minecraft (and the entire system once) crashed

**Logs and screenshots**
https://paste.ee/p/QASSU

If you don't have M1 hardware I am more than willing to test things for you.

The log shows a bunch of mods, but I tried it with no other mods at all with the same exact result.

I used these instructions to make Minecraft run natively on M1: https://github.com/yusefnapora/m1-multimc-hack

The game works perfectly fine on basic.

I have the 8 gb ram model of the 2021 M1 Macbook Pro.

**Comment by agentk** [comment]
@FourInchKnife I have the same issue on M1 MacBook. I thought it was just me doing something wrong. I’ve also noticed with on the built in display it only runs at 1x res (on basic).

**Comment by ryleu** [comment]
> @FourInchKnife I have the same issue on M1 MacBook. I thought it was just me doing something wrong. I’ve also noticed with on the built in display it only runs at 1x res (on basic).

What do you mean by 1x resolution? It looks fine for me at least...

Did you have to use yusefnapora’s LWJGL hack?

**Comment by agentk** [comment]
**System Informatinon**

* Minecraft Version: 1.17
* Canvas Version: 1.17-1.0.1641 & 1.17-1.0.1684
* Operating System: MacOS v11.4 (Big Sur) on ARM
* Graphics Card: Apple M1
* fabric-api-0.36.0+1.17
* Java: Zulu 16.0.1+9

**Logs and screenshots**
https://paste.ee/p/2D3N9

I'm using the wrapper form yusefnapora/m1-multimc-hack along with the lwjgl overrides.

Screenshots of the two resolutions are at:
https://imgur.com/a/RLtWcLx

First screenshot is Canvas Basic and rendering at 1440x900 (and upscaled for the laptop display). You notice the jagged edges most on things like glass.
Second screenshot is the same world reloaded without any mods. It's rendering at 2880x1800. Ignore the difference in GUI scaling I forgot to match the setting. It renders edges a lot smoother.

**Comment by ryleu** [comment]
> 
> 
> **System Informatinon**
> 
>     * Minecraft Version: 1.17
> 
>     * Canvas Version: 1.17-1.0.1641 & 1.17-1.0.1684
> 
>     * Operating System: MacOS v11.4 (Big Sur) on ARM
> 
>     * Graphics Card: Apple M1
> 
>     * fabric-api-0.36.0+1.17
> 
>     * Java: Zulu 16.0.1+9
> 
> 
> **Logs and screenshots**
> https://paste.ee/p/2D3N9
> 
> I'm using the wrapper form yusefnapora/m1-multimc-hack along with the lwjgl overrides.
> 
> Screenshots of the two resolutions are at:
> https://imgur.com/a/RLtWcLx
> 
> First screenshot is Canvas Basic and rendering at 1440x900 (and upscaled for the laptop display). You notice the jagged edges most on things like glass.
> Second screenshot is the same world reloaded without any mods. It's rendering at 2880x1800. Ignore the difference in GUI scaling I forgot to match the setting. It renders edges a lot smoother.

Have you tried disabling the reduce resolution on macs option? This seems to me like an unrelated issue...

**Comment by agentk** [comment]
> Have you tried disabling the reduce resolution on macs option? This seems to me like an unrelated issue...

Haha. You nailed it. I never even thought that there would be an option to toggle it.

Much better now 😄 
https://i.imgur.com/0vJkzNZ.png

Now I'm down to the crash on joining a world with standard pipeline same as you raised.

**Comment by ryleu** [comment]
@agentk Glad I could help :)

It seems to be a GPU hang issue. It is possible that this is the M1 LWJGL hack’s fault?

**Comment by spiralhalo** [comment]
Does anyone know about the equivalent vanilla bug?

https://bugs.mojang.com/browse/MC-218825

It was fixed in the snapshots. Someone vanilla savvy might be able to pinpoint the fix.

**Comment by ryleu** [comment]
> Does anyone know about the equivalent vanilla bug?
> 
> https://bugs.mojang.com/browse/MC-218825
> 
> It was fixed in the snapshots. Someone vanilla savvy might be able to pinpoint the fix.

I think that might be it exactly. I’m going to get a dump of my issue and compare them. If I remember correctly, doesn’t the standard pipeline also set the game to Fabulous graphics?

**Comment by spiralhalo** [comment]
Yes, the standard pipeline is the Canvas equivalent of Fabulous graphics.

**Comment by ryleu** [comment]
Good to know. I’m not particularly vanilla savvy, but I can open two instances of Intellij and try to look for changes in that area if need be.
